populate_analytics.js

```diff
@@ -0,0 +1,283 @@
+
+const cardUses = {};
+const cardSizeUses = {
+    size180: {},
+    size360: {},
+    size450: {},
+    size540: {},
+    size720: {},
+    pauper: {},
+    legacy: {},
+    modern: {},
+    standard: {},
+    vintage: {},
+};
+
+//global cube stats
+const cubeCounts = {
+    total: 0,
+    size180: 0,
+    size360: 0,
+    size450: 0,
+    size540: 0,
+    size720: 0,
+    pauper: 0,
+    legacy: 0,
+    modern: 0,
+    standard: 0,
+    vintage: 0,
+}
+
+const correlationIndex = {};
+const correlations = [];
+
+//use correlationIndex for index
+const cubesWithCard = [];
+
+function createCorrelations() {
+    const totalCards = carddb.cardnames.length;
+    for (var i = 0; i < totalCards; i ++) {
+        correlationIndex[carddb.cardnames[i].toLowerCase()] = i;
+        correlations.push([]);
+        cubesWithCard.push([]);
+        for (var j = 0; j < totalCards; j ++) {
+            correlations[i].push(0);
+        }
+        if((i+1)%100==0){
+            console.log('Finished: ' + (i+1) + ' of ' + totalCards + ' correlations.');
+        }
+    }
+    console.log('Finish init of correlation matrix.')
+}
+
+function attemptIncrement(obj, propname) {
+    if(!obj[propname]) {
+        obj[propname] = 0;
+    }
+    obj[propname]++; 
+
+}
+
+async function processDeck(deck) {
+    if(deck.playerdeck && deck.playerdeck.length > 0) {
+        //flatten array
+        const deckCards = [];
+        deck.playerdeck.forEach(function(col, index) {
+            if(Array.isArray(col)) {
+                col.forEach(function(row, index2) {
+                    deckCards.push(row.details.name.toLowerCase().normalize('NFD')
+                    .replace(/[\u0300-\u036f]/g, '')
+                    .trim());
+                });
+            } else {
+                deckCards.push(col.details.name.toLowerCase().normalize('NFD')
+                .replace(/[\u0300-\u036f]/g, '')
+                .trim());
+            }
+        });
+
+        for(let i = 0; i < deckCards.length; i++) {
+            //could be an invalid card
+            if(correlationIndex[deckCards[i]] && !basics.includes(deckCards[i])) {
+                for(let j = i+1; j < deckCards.length; j++) {
+                    if(!basics.includes(deckCards[j])) {
+                        try {
+                            correlations[correlationIndex[deckCards[j]]]
+                                        [correlationIndex[deckCards[i]]]++;
+                            correlations[correlationIndex[deckCards[j]]]
+                                        [correlationIndex[deckCards[i]]]++;
+                        } catch(err) {
+                            console.log(deckCards[i] + ' or ' + deckCards[j] + ' cannot be indexed.');
+                        }
+                    }
+                }
+            }
+        }
+    }
+}
+
+async function processCube(cube) {
+
+    let cubeSizeDict = cardSizeUses.size180;
+    let cubeLegalityDict = cardSizeUses.vintage;
+
```

Comment:
I wonder if there should be some sort of looser/fuzzier boundary here? Are people sticklers for sizes or is close enough better for categorization? If looser, just by  a absolute amount? 10? Or percentage?
```js
let fuzzyBoundary = 1.05; // allow for sizes within 5% of card count
if (cube.card_count <= Math.floor(180 * fuzzyBoundary)) {
```

Reply:
the actual ranges are:
![image](https://user-images.githubusercontent.com/28238025/70063433-be0cdf80-15b5-11ea-85c4-6c2baefa7854.png)